### PR TITLE
Add support for Rollbar exception handling

### LINKED ISSUE #1103: Setup Rollbar
Setup an environment based exception logging service using our Rollbar account.

### PR DESCRIPTION
**Pull request by hectcastro**
Add Rollbar exception handling support for Django, Celery, and Windshaft. In addition, ensure that the server side token and stack type are propagated to new AWS instances via `cloud-config`.

Attempts to resolve #1103.

---

**Testing**

Testing involves triggering exceptions within Django, Celery, and Windshaft on staging, because it is difficult to make the Django production settings file work locally.

First, ensure that there is a setting within your `default.yml` configuration file with `RollbarServerSideAccessToken` set to the value of `post_server_item` within the **Project Access Tokens** Rollbar UI.

Then, use the following AMI IDs to test this branch:

```yaml
AppServerAMI: 'ami-c34d6ea9'
TileServerAMI: 'ami-174c6f7d'
WorkerAMI: 'ami-344e6d5e'
```

### CODE REVIEW

**Comment by hectcastro** [comment]
@azavea-bot rebuild

**Comment by mmcfarland** [comment]
Was able to cause and see logged errors for Celery and Django errors:

screenshot from 2016-01-28 22 11 26

However, the Windshaft server on the referenced AMI wouldn't start (and was failing health checks) with `rollbar` module not found errors.  I see the package added to the `package.json` file, but it wasn't in `node_modules`.  I manually `npm install`ed it to test errors, but despite successfully inserting bad code to any of the `server.js` methods, no errors were logged.

screenshot from 2016-01-28 22 03 56

**Comment by hectcastro** [comment]
Do you recall launching the stack with `--activate-dns`?

**Comment by mmcfarland** [comment]
I didn't and in fact I used the ELB DNS name to access the test stack while I left the regular staging stack up.  Didn't occur to me that Rollbar might be filtering on events from a defined domain.

**Comment by hectcastro** [comment]
How'd you try to trigger that error?

**Comment by mmcfarland** [comment]
I simulated some bad code in `beforeTileRender`, `afterTileRender` and `req2parameters` in server.js.  I think I tried to access something off of an undefined variable, `x.ghost_mouse` or similar.

**Comment by hectcastro** [comment]
I think I got Rollbar to capture the ghost mouse:

```diff
diff --git a/src/tiler/server.js b/src/tiler/server.js
index 3237fc7..77ce667 100644
--- a/src/tiler/server.js
+++ b/src/tiler/server.js
@@ -84,6 +84,8 @@ var config = {
 
     afterTileRender: function(req, res, tile, headers, callback) {
         try {
+            var x = x.ghost_mouse;
+
             // Complete render pipline first, add cache header for
             // 30 days
             headers['Cache-Control'] = 'max-age=2592000';
```

banners_and_alerts_and__12_typeerror__cannot_read_property__ghost_mouse__of_undefined_-_mmw_-_rollbar

---

Mind taking this for a spin again? If you populate `/etc/mmw.d/env/ROLLBAR_SERVER_SIDE_ACCESS_TOKEN` on the `tiler` with the Rollbar access token (and restart the service), you should be able to test this locally.

**Comment by hectcastro** [comment]
@azavea-bot rebuild

**Comment by mmcfarland** [comment]
+1, Windshaft exceptions recorded and npm packages on `tiler` were installed correctly.

screenshot from 2016-02-01 08 53 51

File: src/tiler/package.json [hunk 1]
```diff
@@ -4,8 +4,9 @@
   "description": "",
   "main": "server.js",
   "dependencies": {
-    "windshaft": "0.36.0",
-    "aws-sdk": "2.1.45"
+    "aws-sdk": "2.1.45",
+    "rollbar": "^0.5.12",
```
Comment (caseypt) [inline src/tiler/package.json]: Just making sure we want the carrot here. It may not matter since we have a shrinkwrap file.

Comment (hectcastro) [inline src/tiler/package.json]: After looking it up, `^` appears to allow changes to `MINOR` version, whereas `~` only allows updates to `PATCH`. I think we want to only allow flexibility with `PATCH`.

**Pull request merged**